airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py

```diff
@@ -116,7 +116,10 @@ def read(self) -> Iterable[Record]:
             if finished_partitions and all(partitions_to_done.values()):
                 # All partitions were generated and process. We're done here
                 break
-        self._check_for_errors(futures)
+            self._check_for_errors(futures)
```

Comment:
check for errors at each iterations instead of waiting until the end of the sync